helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -430,7 +431,7 @@ public String toString() {
   @Override
   protected Map<String, ResourceAssignment> computeBestPossibleAssignment(
       ResourceControllerDataProvider clusterData, Map<String, Resource> resourceMap,
-      Set<String> activeNodes, CurrentStateOutput currentStateOutput)
+      Set<String> activeNodes, CurrentStateOutput currentStateOutput, RebalanceAlgorithm algorithm)
```

Comment:
Is changing the signature of the protected method safe? Will it not break backward-compatibility? 

This verifier is a public class being used by users?

Reply:
No worry, if you look carefully, this is newly added. Still in the branch.
But it could be better if I just move these 2 classes to the verifier and mark them as private classes. So the user won't use them.